dev-python/bokeh/bokeh-2.0.2.ebuild

```diff
@@ -1,59 +1,61 @@
 # Copyright 1999-2020 Gentoo Authors
 # Distributed under the terms of the GNU General Public License v2
 
-EAPI=6
+EAPI=7
 
-PYTHON_COMPAT=( python3_6 )
+DISTUTILS_USE_SETUPTOOLS=rdepend
+
+# upstream hasn't tested python 3.8 fully
+PYTHON_COMPAT=( python3_{6,7} )
 
 inherit distutils-r1
 
 DESCRIPTION="Statistical and interactive HTML plots for Python"
-HOMEPAGE="https://bokeh.pydata.org/en/latest/
+HOMEPAGE="https://bokeh.org/
 	https://github.com/bokeh/bokeh
 	https://pypi.org/project/bokeh/"
 SRC_URI="mirror://pypi/${P:0:1}/${PN}/${P}.tar.gz"
 
 LICENSE="BSD"
 SLOT="0"
-KEYWORDS="amd64 x86 ~amd64-linux ~x86-linux"
-IUSE="examples test"
-RESTRICT="!test? ( test )"
+KEYWORDS="~amd64 ~x86 ~amd64-linux ~x86-linux"
+IUSE="examples"
+
+distutils_enable_tests pytest
 
-RDEPEND="
+# upstream authoritative dependencies
+# https://github.com/bokeh/bokeh/blob/master/conda.recipe/meta.yaml
+DEPEND="
 	dev-python/jinja[${PYTHON_USEDEP}]
+	dev-python/markupsafe[${PYTHON_USEDEP}]
 	dev-python/numpy[${PYTHON_USEDEP}]
+	dev-python/packaging[${PYTHON_USEDEP}]
+	dev-python/pillow[${PYTHON_USEDEP}]
+	dev-python/pyparsing[${PYTHON_USEDEP}]
 	dev-python/python-dateutil[${PYTHON_USEDEP}]
 	dev-python/pyyaml[${PYTHON_USEDEP}]
 	dev-python/requests[${PYTHON_USEDEP}]
 	dev-python/six[${PYTHON_USEDEP}]
-	>=www-servers/tornado-4.3[${PYTHON_USEDEP}]
+	dev-python/typing-extensions[${PYTHON_USEDEP}]
+	>=www-servers/tornado-5[${PYTHON_USEDEP}]
 "
-DEPEND="
-	dev-python/jinja[${PYTHON_USEDEP}]
-	dev-python/numpy[${PYTHON_USEDEP}]
-	dev-python/python-dateutil[${PYTHON_USEDEP}]
-	dev-python/pyyaml[${PYTHON_USEDEP}]
-	dev-python/packaging[${PYTHON_USEDEP}]
-	dev-python/setuptools[${PYTHON_USEDEP}]
-	dev-python/six[${PYTHON_USEDEP}]
-	>=net-libs/nodejs-4.1
-	>=www-servers/tornado-4.3[${PYTHON_USEDEP}]
+RDEPEND="${DEPEND}"
+BDEPEND="
+	${DEPEND}
+	sys-apps/ripgrep[pcre]
 	test? (
 		dev-python/beautifulsoup:4[${PYTHON_USEDEP}]
 		dev-python/boto[${PYTHON_USEDEP}]
 		dev-python/colorama[${PYTHON_USEDEP}]
 		dev-python/flake8[${PYTHON_USEDEP}]
 		dev-python/pandas[${PYTHON_USEDEP}]
 		dev-python/pytest[${PYTHON_USEDEP}]
-		dev-python/flexx[${PYTHON_USEDEP}]
 		dev-python/mock[${PYTHON_USEDEP}]
 		dev-python/nbformat[${PYTHON_USEDEP}]
-		dev-python/pillow[${PYTHON_USEDEP}]
-		dev-python/pytest[${PYTHON_USEDEP}]
-		dev-python/pytest-cov[${PYTHON_USEDEP}]
 		dev-python/pytest-rerunfailures[${PYTHON_USEDEP}]
 		dev-python/pytest-xdist[${PYTHON_USEDEP}]
 		dev-python/selenium[${PYTHON_USEDEP}]
+		>=net-libs/nodejs-12[npm]
 		sci-libs/scipy[${PYTHON_USEDEP}]
```

Comment:
Still missing *flaky* and *networkx* and  from this list. And you know what, I don't believe you got tests to work because it attempts to download bunch of stuff from the internet. You need to disable or remove these kind of dangerous tests. 

Working python tests are pretty much the guarantee that this package works.

Reply:
I haven't managed to get the tests to work yet, I'm still trying to figure out their code.  
I suggest ignoring this PR for now.  
I'm going to bug upstream about this.

Reply:
done